### ISSUE TITLE
convert should recover orphaned blocks

### ISSUE THREAD

**Issue by boredzo** [boredzo/impluse-hfs#59]
A volume might have allocation blocks that are marked in the volume bitmap as in use but not actually assigned to an extent. There might be data in these blocks that would be lost if a conversion didn't take care to preserve them.

One way to do this might be to create a new file in the destination volume and copy the orphaned blocks as that file's contents.

**Comment by boredzo** [comment]
Note that recovered data might be a data fork or a resource fork. The copied blocks should be assigned to both forks; if the data is a resource map, then it will at least be viewable in ResEdit and possibly be usable if the file's correct type can be guessed.

(Of course, it might also be a data fork *and* a resource fork, but we can't help that since we can't tell where anything was supposed to begin or end among the recovered blocks.)

**Comment by boredzo** [comment]
I have a working implementation of this, but it's too sensitive and not specific enough: It also creates the recovery file on volumes that have no orphaned blocks, because the creation of the recovery file has to happen before the catalog is finalized (and its space allocated), but no real files have been copied (and their blocks visited) yet, so the orphan check at that time thinks almost everything is orphaned.

The existing “how many blocks haven't been visited yet?” check (which was originally a debugging method) won't work at that point in the process; I'll need to implement a new, separate traversal of the source catalog's leaf row, marking off blocks in a separate copy of the VBM.

**Comment by boredzo** [comment]
Whoops, I already filed this. Duplicate of #58.
